upstairs/src/lib.rs

```diff
@@ -4505,8 +4468,22 @@ impl Downstairs {
                     if jobs_completed_ok == 1 {
                         assert!(job.data.is_none());
                         assert!(job.read_response_hashes.is_empty());
+
+                        job.read_response_hashes = read_data
+                            .iter()
+                            .map(|read_response| {
+                                // Each entry in a validated list of read
+                                // responses will have either an empty list of
+                                // hashes (meaning the data block was all
+                                // zeros), or one valid context (because the
+                                // rest were filtered out). Turn this into an
+                                // Option here.
+                                read_response.hashes().first().copied()
```

Comment:
We could use the hypothetical `first_hash()` function here too!

Reply:
added in 6eb6d22 :)